source/application/SGM/SGMAppController.cpp

```diff
@@ -147,38 +148,31 @@ void SGMAppController::resizeToMinimum()
 	mw_->resize(mw_->minimumSizeHint());
 }
 
-void SGMAppController::onUserConfigurationLoadedFromDb()
-{
-	AMXRFDetector *detector = SGMBeamline::sgm()->amptekSDD1();
-
-	foreach (AMRegionOfInterest *region, sgmUserConfiguration_->regionsOfInterest()){
-		detector->addRegionOfInterest(region);
-		xasScanConfiguration_->addRegionOfInterest(region);
-		lineScanConfiguration_->addRegionOfInterest(region);
-		mapScanConfiguration_->addRegionOfInterest(region);
-	}
-
-	// This is connected here because we want to listen to the detectors for updates, but don't want to double add regions on startup.
-	connect(detector, SIGNAL(addedRegionOfInterest(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestAdded(AMRegionOfInterest*)));
-	connect(detector, SIGNAL(removedRegionOfInterest(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestRemoved(AMRegionOfInterest*)));
-}
-
 void SGMAppController::onRegionOfInterestAdded(AMRegionOfInterest *region)
 {
-	sgmUserConfiguration_->addRegionOfInterest(region);
+	userConfiguration_->addRegionOfInterest(region);
 	xasScanConfiguration_->addRegionOfInterest(region);
 	lineScanConfiguration_->addRegionOfInterest(region);
 	mapScanConfiguration_->addRegionOfInterest(region);
 }
 
 void SGMAppController::onRegionOfInterestRemoved(AMRegionOfInterest *region)
 {
-	sgmUserConfiguration_->removeRegionOfInterest(region);
+	userConfiguration_->removeRegionOfInterest(region);
 	xasScanConfiguration_->removeRegionOfInterest(region);
 	lineScanConfiguration_->removeRegionOfInterest(region);
 	mapScanConfiguration_->removeRegionOfInterest(region);
 }
 
+void SGMAppController::onRegionOfInterestBoundingRangeChanged(AMRegionOfInterest *region)
+{
+	userConfiguration_->setRegionOfInterestBoundingRange(region);
+//	xasScanConfiguration_->setRegionOfInterestBoundingRange(region);
```

Comment:
Good question! First this is the SGM codes, which we won't care about it at this moment; second, SGM doesn't have the slot for the ROI bounding range changed. I don't know whether they want it or not. That's why I added those codes but commented off. 

At least in the future, if SGM is using Acuqman again, and if they have questions on the ROI bounding range changed, we know we thought about this.